Remove CSSStyleDeclaration#isPropertyImplicit from the web exposed API

darinadler opened the pull request:
#### d20f6c0295260f8adc679dde22c8b9c00c01ad93
<pre>
Remove CSSStyleDeclaration#isPropertyImplicit from the web exposed API
<a href="https://bugs.webkit.org/show_bug.cgi?id=217625">https://bugs.webkit.org/show_bug.cgi?id=217625</a>
rdar://problem/70447880

Reviewed by Tim Nguyen.

* LayoutTests/css3/color-filters/color-filter-exposed-if-disabled-expected.txt: Updated.
* LayoutTests/css3/color-filters/color-filter-exposed-if-disabled.html: Removed test of
isPropertyImplicit.

* LayoutTests/css3/color-filters/color-filter-exposed-if-enabled-expected.txt: Updated.
* LayoutTests/css3/color-filters/color-filter-exposed-if-enabled.html: Removed test of
isPropertyImplicit.

* LayoutTests/fast/css/font-shorthand-expected.txt: Updated.
* LayoutTests/fast/css/font-shorthand.html: Removed code that dumps the name of the shorthand
and the result of isPropertyImplicit.

* LayoutTests/fast/css/implicit-property-restore-expected.txt: Removed.
* LayoutTests/fast/css/implicit-property-restore.html: Removed.

* LayoutTests/fast/css/style-enumerate-properties-expected.txt: Updated.
* LayoutTests/fast/css/style-enumerate-properties.html: Removed isPropertyImplicit.

* LayoutTests/fast/inspector-support/style-expected.txt: Removed.
* LayoutTests/fast/inspector-support/style.html: Removed.

* LayoutTests/platform/win/fast/css/font-shorthand-expected.txt: Updated.

* LayoutTests/platform/win/fast/inspector-support/style-expected.txt: Removed.

* Source/WebCore/css/CSSStyleDeclaration.idl: Removed isPropertyImplicit.

* Source/WebCore/css/PropertySetCSSStyleDeclaration.cpp:
(WebCore::PropertySetCSSStyleDeclaration::isPropertyImplicit): Removed unneeded check of isExposed
since the clients of this function are now all internal and not web-exposed.

Canonical link: <a href="https://commits.webkit.org/256633@main">https://commits.webkit.org/256633@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eb970cfcd2ad3a8614d62d3bd342f18699082797

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/96389 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/5643 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/29453 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/105932 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/166277 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/100372 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/5812 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/34389 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/88768 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/102658 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/102060 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/4323 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/82990 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/31311 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/86165 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/88057 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/74199 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/40119 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/19551 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37793 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/20938 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4602 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/2968 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/43509 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/641 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/40207 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/PropertySetCSSStyleDeclaration.cpp:
```diff
@@ -238,10 +238,7 @@ String PropertySetCSSStyleDeclaration::getPropertyShorthand(const String& proper
 {
-    CSSPropertyID propertyID = cssPropertyID(propertyName);
-    if (!isExposed(propertyID))
-        return false;
-    return m_propertySet->isPropertyImplicit(propertyID);
+    return m_propertySet->isPropertyImplicit(cssPropertyID(propertyName));
 }
```
I don't see any other usages of this method in the codebase. Can we remove it altogether?

nt1m replied:
Oh nevermind, it is used: https://searchfox.org/wubkat/search?q=symbol:_ZN7WebCore19CSSStyleDeclaration18isPropertyImplicitERKN3WTF6StringE&redirect=false

darinadler replied:
This function is still used. The uses are in these functions:
```
InspectorStyle::shorthandValue
InspectorStyle::styleWithProperties
webkit_dom_css_style_declaration_is_property_implicit
-[DOMCSSStyleDeclaration isPropertyImplicit:]
```
If someone takes care of all of these, then we can remove the function altogether. I don’t know the inspector well enough to know exactly how to remove the first two. The others could likely just be changed to always return `false`.

nt1m replied:
Looking at the remaining usages, I'm tempted to think it's ok to remove them.

One is to skip implicit props from showing in the WI UI, the other one is GTK API. Given how we use implicit properties nowadays, I'm not sure it still makes sense to skip them. cc @patrickangle 

Anyway, this doesn't need to be done here.

darinadler replied:
Once we remove the function we can also remove the flag, so it will be a nice cleanup!

rniwa commented on Source/WebCore/css/CSSStyleDeclaration.idl:
```diff
@@ -40,6 +40,5 @@ typedef USVString CSSOMString;
 
     // Non-standard.
     DOMString? getPropertyShorthand(optional DOMString propertyName);
-    boolean isPropertyImplicit(optional DOMString propertyName);
```
Is this change Web compatible??? It seems plausible that there is a bunch of iOS specific content that's relying on this. A quick search on GitHub yields about 155 entries:
https://github.com/search?q=isPropertyImplicit(%20language%3AJavaScript&type=code

nt1m replied:
Looking at the entries, they're either:
- WebInspector instance in WebKit forks
- Chrome DevTools forks (repos named devtools-frontend or inspector)
- Epiphany
- Old forks of Chromium that still have WebKit

Seems safe to remove to me?

nt1m replied:
Perhaps @karlcow wants to take a look.

darinadler committed:
Remove CSSStyleDeclaration#isPropertyImplicit from the web exposed API
https://bugs.webkit.org/show_bug.cgi?id=217625
rdar://problem/70447880

Reviewed by Tim Nguyen.

* LayoutTests/css3/color-filters/color-filter-exposed-if-disabled-expected.txt: Updated.
* LayoutTests/css3/color-filters/color-filter-exposed-if-disabled.html: Removed test of
isPropertyImplicit.

* LayoutTests/css3/color-filters/color-filter-exposed-if-enabled-expected.txt: Updated.
* LayoutTests/css3/color-filters/color-filter-exposed-if-enabled.html: Removed test of
isPropertyImplicit.

* LayoutTests/fast/css/font-shorthand-expected.txt: Updated.
* LayoutTests/fast/css/font-shorthand.html: Removed code that dumps the name of the shorthand
and the result of isPropertyImplicit.

* LayoutTests/fast/css/implicit-property-restore-expected.txt: Removed.
* LayoutTests/fast/css/implicit-property-restore.html: Removed.

* LayoutTests/fast/css/style-enumerate-properties-expected.txt: Updated.
* LayoutTests/fast/css/style-enumerate-properties.html: Removed isPropertyImplicit.

* LayoutTests/fast/inspector-support/style-expected.txt: Removed.
* LayoutTests/fast/inspector-support/style.html: Removed.

* LayoutTests/platform/win/fast/css/font-shorthand-expected.txt: Updated.

* LayoutTests/platform/win/fast/inspector-support/style-expected.txt: Removed.

* Source/WebCore/css/CSSStyleDeclaration.idl: Removed isPropertyImplicit.

* Source/WebCore/css/PropertySetCSSStyleDeclaration.cpp:
(WebCore::PropertySetCSSStyleDeclaration::isPropertyImplicit): Removed unneeded check of isExposed
since the clients of this function are now all internal and not web-exposed.

Canonical link: https://commits.webkit.org/256633@main

karlcow reviewed (commented):
looks Web Compatible.
